### ISSUE TITLE
Add Fortran backend

### ISSUE THREAD

**Issue by Thirumalai-Shaktivel** [lfortran/lfortran#2410]
Print any ASR to Fortran code

**Comment by Thirumalai-Shaktivel** [comment]
I had two ideas:
1. Convert `ASR to AST` and use `ast_to_src()` to print the Fortran Source code.
2. Convert ASR directly to source code.

I lean more towards option 2, as it can be used by other frontends as well.

**Comment by certik** [comment]
Yes, as we discussed, we should go with 2., since it will also be easier to write and maintain, similar to the C backend, but even simpler, since ASR maps better to Fortran.

**Comment by Thirumalai-Shaktivel** [comment]
TODO
- [ ] Add color to the source code
- [ ] Visit symbols in Module symtab
- [ ] Create a separate function or macro to visit symbols in symtab and handling body for the Nodes like Function, Module, ...
- [ ] Add Fortran to all the tests in the integration_tests CMakeLists (this would fix all the above)
